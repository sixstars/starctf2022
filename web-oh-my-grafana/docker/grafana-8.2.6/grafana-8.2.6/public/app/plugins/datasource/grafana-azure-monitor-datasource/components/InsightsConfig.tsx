import React, { PureComponent } from 'react';
import { InlineFormLabel, Button, LegacyForms, Alert } from '@grafana/ui';
const { Input } = LegacyForms;
import { AzureDataSourceSettings, AzureDataSourceJsonData, AzureDataSourceSecureJsonData } from '../types';

export interface Props {
  options: AzureDataSourceSettings;
  onUpdateJsonDataOption: (
    key: keyof AzureDataSourceJsonData
  ) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => void;
  onUpdateSecureJsonDataOption: (
    key: keyof AzureDataSourceSecureJsonData
  ) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => void;
  onResetOptionKey: (key: keyof AzureDataSourceSecureJsonData) => void;
}
export class InsightsConfig extends PureComponent<Props> {
  onAppInsightsResetApiKey = () => {
    this.props.onResetOptionKey('appInsightsApiKey');
  };

  render() {
    const { options, onUpdateJsonDataOption, onUpdateSecureJsonDataOption } = this.props;
    return (
      <>
        <h3 className="page-heading">Azure Application Insights</h3>
        <Alert severity="info" title="Application Insights credentials are deprecated">
          Configure using Azure AD App Registration above and update existing queries to use Metrics or Logs.
        </Alert>
        <div className="gf-form-group">
          {options.secureJsonFields.appInsightsApiKey ? (
            <div className="gf-form-inline">
              <div className="gf-form">
                <InlineFormLabel className="width-12">API Key</InlineFormLabel>
                <Input className="width-25" placeholder="configured" disabled={true} />
              </div>
              <div className="gf-form">
                <div className="max-width-30 gf-form-inline">
                  <Button
                    variant="secondary"
                    type="button"
                    onClick={this.onAppInsightsResetApiKey}
                    disabled={this.props.options.readOnly}
                  >
                    reset
                  </Button>
                </div>
              </div>
            </div>
          ) : (
            <div className="gf-form-inline">
              <div className="gf-form">
                <InlineFormLabel className="width-12">API Key</InlineFormLabel>
                <div className="width-15">
                  <Input
                    className="width-30"
                    placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
                    value={options.secureJsonData!.appInsightsApiKey || ''}
                    onChange={onUpdateSecureJsonDataOption('appInsightsApiKey')}
                    disabled={this.props.options.readOnly}
                  />
                </div>
              </div>
            </div>
          )}
          <div className="gf-form-inline">
            <div className="gf-form">
              <InlineFormLabel className="width-12">Application ID</InlineFormLabel>
              <div className="width-15">
                <Input
                  className="width-30"
                  value={options.jsonData.appInsightsAppId || ''}
                  onChange={onUpdateJsonDataOption('appInsightsAppId')}
                  disabled={this.props.options.readOnly}
                />
              </div>
            </div>
          </div>
        </div>
      </>
    );
  }
}

export default InsightsConfig;
