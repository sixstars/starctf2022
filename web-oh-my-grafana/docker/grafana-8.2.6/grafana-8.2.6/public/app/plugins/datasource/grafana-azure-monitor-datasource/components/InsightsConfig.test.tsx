import React from 'react';
import { shallow } from 'enzyme';
import InsightsConfig, { Props } from './InsightsConfig';
import { Button, LegacyForms } from '@grafana/ui';
const { Input } = LegacyForms;

const setup = (propOverrides?: object) => {
  const props: Props = {
    options: {
      id: 21,
      uid: 'x',
      orgId: 1,
      name: 'Azure Monitor-10-10',
      type: 'grafana-azure-monitor-datasource',
      typeLogoUrl: '',
      typeName: 'Azure',
      access: 'proxy',
      url: '',
      password: '',
      user: '',
      database: '',
      basicAuth: false,
      basicAuthUser: '',
      basicAuthPassword: '',
      withCredentials: false,
      isDefault: false,
      secureJsonFields: {},
      jsonData: {
        cloudName: '',
        subscriptionId: '',
      },
      secureJsonData: {},
      version: 1,
      readOnly: false,
    },
    onUpdateJsonDataOption: jest.fn(),
    onUpdateSecureJsonDataOption: jest.fn(),
    onResetOptionKey: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<InsightsConfig {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should disable insights api key input', () => {
    const wrapper = setup({
      options: {
        secureJsonFields: {
          appInsightsApiKey: true,
        },
        jsonData: {
          appInsightsAppId: 'cddcc020-2c94-460a-a3d0-df3147ffa792',
        },
        secureJsonData: {
          appInsightsApiKey: 'e7f3f661-a933-4b3f-8176-51c4f982ec48',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should enable insights api key input', () => {
    const wrapper = setup({
      options: {
        secureJsonFields: {
          appInsightsApiKey: false,
        },
        jsonData: {
          appInsightsAppId: 'cddcc020-2c94-460a-a3d0-df3147ffa792',
        },
        secureJsonData: {
          appInsightsApiKey: 'e7f3f661-a933-4b3f-8176-51c4f982ec48',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should disable buttons and inputs', () => {
    const wrapper = setup({
      options: {
        secureJsonFields: {
          appInsightsApiKey: true,
        },
        jsonData: {
          appInsightsAppId: 'cddcc020-2c94-460a-a3d0-df3147ffa792',
        },
        secureJsonData: {
          appInsightsApiKey: 'e7f3f661-a933-4b3f-8176-51c4f982ec48',
        },
        readOnly: true,
      },
    });
    const buttons = wrapper.find(Button);
    const inputs = wrapper.find(Input);
    buttons.forEach((b) => expect(b.prop('disabled')).toBe(true));
    inputs.forEach((i) => expect(i.prop('disabled')).toBe(true));
  });
});
