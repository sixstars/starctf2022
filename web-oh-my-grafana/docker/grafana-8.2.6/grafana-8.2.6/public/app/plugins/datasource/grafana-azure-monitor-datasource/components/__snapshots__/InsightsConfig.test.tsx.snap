// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should disable insights api key input 1`] = `
<Fragment>
  <h3
    className="page-heading"
  >
    Azure Application Insights
  </h3>
  <Alert
    severity="info"
    title="Application Insights credentials are deprecated"
  >
    Configure using Azure AD App Registration above and update existing queries to use Metrics or Logs.
  </Alert>
  <div
    className="gf-form-group"
  >
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          API Key
        </FormLabel>
        <Input
          className="width-25"
          disabled={true}
          placeholder="configured"
        />
      </div>
      <div
        className="gf-form"
      >
        <div
          className="max-width-30 gf-form-inline"
        >
          <Button
            onClick={[Function]}
            type="button"
            variant="secondary"
          >
            reset
          </Button>
        </div>
      </div>
    </div>
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          Application ID
        </FormLabel>
        <div
          className="width-15"
        >
          <Input
            className="width-30"
            value="cddcc020-2c94-460a-a3d0-df3147ffa792"
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Render should enable insights api key input 1`] = `
<Fragment>
  <h3
    className="page-heading"
  >
    Azure Application Insights
  </h3>
  <Alert
    severity="info"
    title="Application Insights credentials are deprecated"
  >
    Configure using Azure AD App Registration above and update existing queries to use Metrics or Logs.
  </Alert>
  <div
    className="gf-form-group"
  >
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          API Key
        </FormLabel>
        <div
          className="width-15"
        >
          <Input
            className="width-30"
            placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
            value="e7f3f661-a933-4b3f-8176-51c4f982ec48"
          />
        </div>
      </div>
    </div>
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          Application ID
        </FormLabel>
        <div
          className="width-15"
        >
          <Input
            className="width-30"
            value="cddcc020-2c94-460a-a3d0-df3147ffa792"
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Render should render component 1`] = `
<Fragment>
  <h3
    className="page-heading"
  >
    Azure Application Insights
  </h3>
  <Alert
    severity="info"
    title="Application Insights credentials are deprecated"
  >
    Configure using Azure AD App Registration above and update existing queries to use Metrics or Logs.
  </Alert>
  <div
    className="gf-form-group"
  >
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          API Key
        </FormLabel>
        <div
          className="width-15"
        >
          <Input
            className="width-30"
            disabled={false}
            placeholder="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <FormLabel
          className="width-12"
        >
          Application ID
        </FormLabel>
        <div
          className="width-15"
        >
          <Input
            className="width-30"
            disabled={false}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
